redaxo/src/addons/structure/plugins/content/lib/article_content_editor.php

```diff
@@ -197,11 +197,32 @@ private function getSliceMenu(rex_sql $artDataSql)
             $header_right .= sprintf('<div class="alert">%s %s</div>', rex_i18n::msg('no_editing_rights'), $moduleName);
         }
 
+        // ----- EXTENSION POINT
+        $slice_actions = [
+            'actions' => $menu_items_actions,
+            'status' => $menu_items_status,
+            'move' => $menu_items_move,
+        ];
+        $slice_actions = rex_extension::registerPoint(new rex_extension_point(
+            'STRUCTURE_CONTENT_SLICE_ACTIONS',
+            $slice_actions,
+            [
+                'context' => $context,
+                'fragment' => $fragment,
+                'article_id' => $this->article_id,
+                'clang' => $this->clang,
+                'ctype' => $sliceCtype,
+                'module_id' => $moduleId,
+                'slice_id' => $sliceId,
+                'perm' => rex::getUser()->getComplexPerm('modules')->hasPerm($moduleId),
+            ]
+        ));
```

Comment:
da die konstruktion mit den vielen parameter nicht so einfach zu konsumieren ist, könnte man überlegen einen namend-extension point zu verwenden.

ich denke ggf. macht es auch sinn den EP `STRUCTURE_CONTENT_SLICE_MENU` als deprecated zu markieren und dessen funktionalität mit in den named-api aufgehen zu lassen, da man dann eine schönere api hätte für das ganze menu statt so zwei gewachsene EPs die array behandeln.

Reply:
> ich denke ggf. macht es auch sinn den EP `STRUCTURE_CONTENT_SLICE_MENU` als deprecated zu markieren und dessen funktionalität mit in den named-EP aufgehen zu lassen, da man dann eine schönere api hätte für das ganze menu statt so zwei gewachsene EPs die array behandeln.

Klingt gut! (auch wenn ich mir die beiden EPs jetzt nicht im Detail angeschaut habe, was die jeweiligen Zwecke sind, und ob sich das gut kombinieren lässt.)